formspree/forms/views.py

```diff
@@ -374,23 +374,96 @@ def unblock_email(email):
-                               text='Please make sure you pass the <i>reCaptcha</i> test before submitting.'), 500
+            title='Unable to unblock email',
+            text='Please make sure you pass the <i>reCaptcha</i> test before '
+                 'submitting.'), 500
 
     elif request.method == 'GET':
         return render_template('forms/unblock_email.html', email=email), 200
 
 
+def request_unconfirm_form(form_id):
+    '''
+    This endpoints triggers a confirmation email that directs users to the
+    GET version of unconfirm_form.
+    '''
+    form = Form.query.get(form_id)
+
+    unconfirm_url = url_for(
+        'unconfirm_form',
+        form_id=form.id,
+        digest=form.unconfirm_digest(),
+        _external=True
+    )
+    send_email(
+        to=form.email,
+        subject='Unsubscribe from form at {}'.format(form.host),
+        html=render_template_string(TEMPLATES.get('unsubscribe-confirmation.html'),
+                                    url=unconfirm_url,
+                                    email=form.email,
+                                    host=form.host),
+        text=render_template('email/unsubscribe-confirmation.txt',
+            url=unconfirm_url,
+            email=form.email,
+            host=form.host),
+        sender=settings.DEFAULT_SENDER,
+    )
+
+    return render_template('info.html',
+        title='Link sent to your address',
+        text="To prevent unwanted form submission losses, we've sent the link "
+             "to finish the unsubscription to {}".format(form.email)), 200
+    
+
 def unconfirm_form(form_id, digest):
     '''
-    We send a digest as the List-Unsubscribe header on every submission.
-    Here we get that digest and handle the unconfirmation request.
+    Here we check the digest for a form and handle the unconfirmation.
+    Also works for List-Unsubscribe triggered POSTs.
+    When GET, give the user the option to unsubscribe from other forms as well.
     '''
     form = Form.query.get(form_id)
-    if form.unconfirm_with_digest(digest):
-        return '', 200
-    else:
-        return abort(401)
+    success = form.unconfirm_with_digest(digest)
+
+    if request.method == 'GET':
+        if success:
+            other_forms = Form.query.filter_by(confirmed=True, email=form.email)
+
+            session['unconfirming'] = form.email
+
+            return render_template('forms/unconfirm.html',
+                other_forms=other_forms,
+                disabled_form=form
+            ), 200
+        else:
+            return render_template('error.html',
+                title='Not a valid link',
+                text='This unconfirmation link is not valid.'), 400
+
+    if request.method == 'POST':
+        if success:
+            return '', 200
+        else:
+            return abort(401)
+
+
+def unconfirm_multiple():
+    unconfirming_for_email = session.get('unconfirming')
+    if not unconfirming_for_email:
```

Comment:
Isn't this insecure because someone could just imitate the cookies?

Reply:
People can imitate cookies, but not sessions. Sessions are signed (or something like that) with the server's secret key such that the server can verify if the data was correctly written by itself and not modified by the client.